### PR TITLE
[SPARK-5678] Convert DataFrame to pandas.DataFrame and Series

### PR DESCRIPTION
**Pull request by davies**
```
pyspark.sql.DataFrame.to_pandas = to_pandas(self) unbound pyspark.sql.DataFrame method
    Collect all the rows and return a `pandas.DataFrame`.

    >>> df.to_pandas()  # doctest: +SKIP
       age   name
    0    2  Alice
    1    5    Bob

pyspark.sql.Column.to_pandas = to_pandas(self) unbound pyspark.sql.Column method
    Return a pandas.Series from the column

    >>> df.age.to_pandas()  # doctest: +SKIP
    0    2
    1    5
    dtype: int64
```

Not tests by jenkins (they depends on pandas)

### CODE REVIEW

**Comment by davies** [comment]
cc @rxin

File: python/pyspark/sql.py [hunk 1]
```diff
@@ -2284,6 +2284,18 @@ def addColumn(self, colName, col):
         """
         return self.select('*', col.alias(colName))
 
+    def to_pandas(self):
+        """
+        Collect all the rows and return a `pandas.DataFrame`.
+
+        >>> df.to_pandas()  # doctest: +SKIP
```
Comment (rxin) [inline python/pyspark/sql.py]: why skipping doctest? Does Jenkins not have this?

Comment (pwendell) [inline python/pyspark/sql.py]: should this be called `to_panda` since a single "panda" is equivalent to a data frame

Comment (davies) [inline python/pyspark/sql.py]: @rxin we have no pandas in jenkins now. We could enable this test once we have it.

@pwendell I think `pandas` is just the project name, `panda` is not meaningful in their context.

**Comment by rxin** [comment]
Thanks. I'm going to merge this.

@pwendell what's the process for installing Pandas on Jenkins machines?

**Comment by rxin** [comment]
I filed https://issues.apache.org/jira/browse/SPARK-5693 to track the test issue.

**Pull request closed without merging**